crates/rome_js_analyze/src/suppression_action.rs

```diff
@@ -0,0 +1,220 @@
+                    );
+                }
+            } else {
+                let mut new_token = first_token_with_newline.clone();
+                if !should_insert_leading_newline {
+                    new_token = new_token.with_leading_trivia([
+                        (TriviaPieceKind::Newline, "\n"),
+                        (
+                            TriviaPieceKind::SingleLineComment,
+                            format!("// {}: suppressed ", suppression_text).as_str(),
+                        ),
+                        (TriviaPieceKind::Newline, "\n"),
+                    ])
+                } else {
+                    new_token = new_token.with_leading_trivia([
+                        (
+                            TriviaPieceKind::SingleLineComment,
+                            format!("// {}: suppressed ", suppression_text).as_str(),
+                        ),
+                        (TriviaPieceKind::Newline, "\n"),
+                    ])
+                };
+                mutation.replace_token_transfer_trivia(first_token_with_newline, new_token);
+            }
+        } else {
+            let mut new_token = first_token_with_newline.clone();
+            if !should_insert_leading_newline {
+                new_token = new_token.with_leading_trivia([
+                    (TriviaPieceKind::Newline, "\n"),
+                    (
+                        TriviaPieceKind::SingleLineComment,
+                        format!("// {}: suppressed ", suppression_text).as_str(),
+                    ),
+                    (TriviaPieceKind::Newline, "\n"),
+                ])
+            } else {
+                new_token = new_token.with_leading_trivia([
+                    (
+                        TriviaPieceKind::SingleLineComment,
+                        format!("// {}: suppressed ", suppression_text).as_str(),
+                    ),
+                    (TriviaPieceKind::Newline, "\n"),
+                ])
+            };
+            mutation.replace_token_transfer_trivia(first_token_with_newline, new_token);
+        }
+    }
+}
+
+/// It checks if the current token has leading trivia newline. If not, it
+/// it peeks the previous token and recursively call itself.
+///
+/// Due to the nature of JSX, sometimes the current token might contain text that contains
+/// some newline. In case that happens, we choose that token.
+///
+/// Due to the nature of JavaScript templates, we also check if the tokens we browse are
+/// `${` and if so, we stop there.
+fn find_first_token_with_newline(token: JsSyntaxToken) -> Option<(JsSyntaxToken, bool)> {
+    let mut current_token = token;
+    let mut should_insert_leading_newline = false;
+    loop {
+        let trivia = current_token.leading_trivia();
+        if trivia.pieces().any(|trivia| trivia.is_newline())
+            || current_token.text_trimmed().contains('\n')
+            || current_token.text_trimmed().contains("\r\n")
+        {
```

Comment:
This test isn't sufficient to handle all JS new line characters:

```
* `\n`
* `\r`
* `\r\n`
```

```suggestion
current_token.text_trimmed().contains(['\n', '\r'])
```


And can you add a test for multiline strings too:

```
"a b\
  cd"
```

I recommend scoping the check for newlines inside of the text to the few token kinds that allow newlines rather than applying it everywhere. The tokens that support newlines in the middle of the text are:

* string literal
* JsxText
* TEMPLATE_CHUNK